### PR TITLE
Tyler Porter 

### PR DESCRIPTION
**Pull request by tylerpporter**

### CODE REVIEW

File: lib/museum.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    matches = []
+    patron.interests.each do |interest|
+      @exhibits.each do |exhibit|
+        if interest == exhibit.name
+          matches << exhibit
+        end
+      end
+    end
+    matches
+  end
```
Comment (tylerpporter) [inline lib/museum.rb]: Ran out of time to refactor. 
`@exhibits.select do |exhibit|
      patron.interests.include?(exhibit.name)
    end`

File: lib/museum.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      patrons = []
+      @patrons.each do |patron|
+        if patron.interests.any? {|interest| interest == exhibit.name}
+          patrons << patron
+        end
+      end
+      by_interest[exhibit] = patrons
+    end
+    by_interest
+  end
```
Comment (tylerpporter) [inline lib/museum.rb]: Ran out of time to refactor.

File: Rakefile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require 'rake/testtask'
```
Comment (memcmahon) [inline Rakefile]: yay rake :)

File: lib/museum.rb (hunk 1 above)
Comment (memcmahon) [reply lib/museum.rb]: Refactor looks great - and also the code that you have totally works, so perfectly fine for a timed challenge 👍

File: lib/museum.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  def admit(patron)
+    @patrons << patron
+  end
+
+  def patrons_by_exhibit_interest
+    by_interest = {}
+    @exhibits.each do |exhibit|
+      patrons = []
+      @patrons.each do |patron|
+        if patron.interests.any? {|interest| interest == exhibit.name}
```
Comment (memcmahon) [inline lib/museum.rb]: Love the use of `any?` With a refactor, you might get rid of this, but it is great to see a good implementation of a less-used enum.

File: lib/museum.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+          patrons << patron
+        end
+      end
+      by_interest[exhibit] = patrons
+    end
+    by_interest
+  end
+
+  def ticket_lottery_contestants(exhibit)
+    patrons_by_exhibit_interest[exhibit].select do |patron|
```
Comment (memcmahon) [inline lib/museum.rb]: Great job reusing the patrons_by_exhibit_interest hash!

File: test/museum_test.rb [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    @dmns.add_exhibit(@imax)
+    @patron_1.add_interest("Dead Sea Scrolls")
+    @patron_1.add_interest("Gems and Minerals")
+    @patron_2.add_interest("IMAX")
+    @patron_3.add_interest("Dead Sea Scrolls")
+    @dmns.admit(@patron_1)
+    @dmns.admit(@patron_2)
+    @dmns.admit(@patron_3)
+    @dmns.stubs(:draw_lottery_winner).returns(@patron_1.name)
+    assert_equal "Bob has won the Dead Sea Scrolls exhibit lottery", @dmns.announce_lottery_winner(@dead_sea_scrolls)
```
Comment (memcmahon) [inline test/museum_test.rb]: Great job ordering your assertions and restubbing to make sure that the stubbed return value doesn't get in the way of either scenario

File: test/museum_test.rb [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    @dmns.add_exhibit(@imax)
+    @patron_1.add_interest("Dead Sea Scrolls")
+    @patron_1.add_interest("Gems and Minerals")
+    @patron_2.add_interest("IMAX")
+    @patron_3.add_interest("Dead Sea Scrolls")
+    @dmns.admit(@patron_1)
+    @dmns.admit(@patron_2)
+    @dmns.admit(@patron_3)
+    @dmns.stubs(:ticket_lottery_contestants).returns([@patron_1])
+    assert_equal "Bob", @dmns.draw_lottery_winner(@dead_sea_scrolls)
```
Comment (memcmahon) [inline test/museum_test.rb]: Not a great use of a stub, because now you are really just testing that stubs work, instead of testing the method itself.

**Pull request closed without merging**